lib/ssl/src/ssl.erl

```diff
@@ -453,130 +601,92 @@ only the [signature_algs](`t:signature_algs/0`) extension is sent.
-                                {handshake, handshake_completion()} |
-                                {cert, cert() | [cert()]} |
-                                {certfile, cert_pem()} |
-                                {key, key()} |
-                                {keyfile, key_pem()} |
-                                {password, key_pem_password()} |
-                                {certs_keys, certs_keys()} |
-                                {ciphers, cipher_suites()} |
-                                {eccs, [named_curve()]} |
-                                {signature_algs, signature_algs()} |
-                                {signature_algs_cert, sign_schemes()} |
-                                {supported_groups, supported_groups()} |
-                                {secure_renegotiate, secure_renegotiation()} |
-                                {keep_secrets, keep_secrets()} |
-                                {depth, allowed_cert_chain_length()} |
-                                {verify_fun, custom_verify()} |
-                                {cert_policy_opts, [policy_opt()]} |
-                                {crl_check, crl_check()} |
-                                {crl_cache, crl_cache_opts()} |
-                                {max_handshake_size, handshake_size()} |
-                                {partial_chain, root_fun()} |
-                                {versions, protocol_versions()} |
-                                {user_lookup_fun, custom_user_lookup()} |
-                                {log_level, logging_level()} |
-                                {log_alert, log_alert()} |
-                                {hibernate_after, hibernate_after()} |
-                                {padding_check, padding_check()} |
-                                {beast_mitigation, beast_mitigation()} |
-                                {ssl_imp, ssl_imp()} |
-                                {session_tickets, session_tickets()} |
-                                {key_update_at, key_update_at()} |
-                                {receiver_spawn_opts, spawn_opts()} |
-                                {sender_spawn_opts, spawn_opts()}.
 
+-doc(#{title => <<"Socket">>}).
 -doc """
-Choose TLS or DTLS protocol for the transport layer security. Defaults to `tls`.
-For DTLS other transports than UDP are not yet supported.
+Error reason for debug purpose should not be matched.
 """.
+-type reason()           :: term().
+
+%% -------------------------------------------------------------------------------------------------------
+
 -doc(#{title =>
-           <<"TLS/DTLS OPTION DESCRIPTIONS - COMMON for SERVER and CLIENT">>}).
--type protocol()                  :: tls | dtls.
+           <<"Client and Server Options">>}).
 -doc """
+Options common to both client and server side.
+
+- **\{protocol, Protocol}** - Choose TLS or DTLS protocol for the transport layer security.
+
+Defaults to `tls`.
```

Comment:
```suggestion
  Defaults to `tls`.
```

You need to indent the body of the bullet list with 2 spaces, otherwise it is treated as a separate paragraph and that is most likely not what you want here. The same mistake is made for many/all bullets in this list and in many other places.

Reply:
Should be fixed now